src/main/java/jenkins/plugins/slack/CommitInfoChoice.java

```diff
@@ -0,0 +1,61 @@
+package jenkins.plugins.slack;
+
+public enum CommitInfoChoice {
+    NONE("nothing about commits") {
+        @Override
+        public boolean showAuthor() {
+            return false;
+        }
+        @Override
+        public boolean showTitle() {
+            return false;
+        }
+    },
+
+    AUTHORS("commit list with authors only") {
+        @Override
+        public boolean showAuthor() {
+            return true;
+        }
+        @Override
+        public boolean showTitle() {
+            return false;
+        }
+    },
+
+    AUTHORS_AND_TITLES("commit list with authors and titles") {
+        @Override
+        public boolean showAuthor() {
+            return true;
+        }
+        @Override
+        public boolean showTitle() {
+            return true;
+        }
+    };
+
+    private final String displayName;
+
+    private CommitInfoChoice(String displayName) {
+        this.displayName = displayName;
+    }
+
+    public abstract boolean showAuthor();
```

Comment:
Add default impls for these two methods that return a boolean, which is passed to the constructor. This can eliminate about 20 LoC.